### DISCUSSION TITLE
Check dependency versions compatibility instead of blindly proposing latest one

### DISCUSSION THREAD

**Discussion by waberc** [renovatebot/renovate#8925] [Suggest an Idea]
**What would you like Renovate to be able to do?**

I am using Renovate in a python project (but this might apply to other technologies) where I have a requirements.txt file I initially created with "pip freeze" to have all pip libraries versions managed. It worked fine for a while now. 

### Actual
However now I am getting troubles with this: 

```txt
requests==2.25.1
idna==2.10
```
Renovate creates me a MR to upgrade to latest idna version `idna==3.1` despite it's not respecting the requirements of `requests` library which needs `idna<3,>=2.5`. My gitlab pipeline that works fine catches this while installing the dependencies: 

Job's logs (while pip install): 

```
ERROR: Cannot install -r requirements.txt (line 18) and idna==3.1 because these package versions have conflicting dependencies.
The conflict is caused by:
    The user requested idna==3.1
    requests 2.25.1 depends on idna<3 and >=2.5
```

### Expected
In my mind it should check the dependency trees to avoid proposing new version that don't respect other requirements

From the doc https://docs.renovatebot.com/python/ I don't see any step where it checks for compatibility of other libraries when checking if a new version is available... An additional step should be added after step 3 in list.

### Workaround
Currently, I might be able to work around this by removing the `idna` library from the requirements.txt file...

**Did you already have any implementation ideas?**
A basic way to do it could be to create a venv and install the libraries to see if that goes successful.

There might be better ways to check the dependency tree versions' compatibility.

**Comment by rarkins** [comment]
If packages are related/dependent, the best thing is that you group them together.

The behaviour you are seeing is intentional and not considered an accident or a bug. There will be times where humans need to resolve update conflicts manually, or choose to ignore an update if resolving it is not possible. It's not luck/chance that you saw the error in your pipelines - the minimum any project should do is at least install things to surface such errors.
